pkg/virt-api/webhooks/mutating-webhook/mutators/vmi-mutator.go

```diff
@@ -90,6 +90,8 @@ func (mutator *VMIsMutator) Mutate(ar *admissionv1.AdmissionReview) *admissionv1
 		mutator.setDefaultResourceRequests(newVMI)
 		mutator.setDefaultGuestCPUTopology(newVMI)
 		mutator.setDefaultPullPoliciesOnContainerDisks(newVMI)
+		mutator.setDefaultArchitecture(newVMI)
```

Comment:
What is the intended behavior of this default?

1. My understanding is that this sets the vmi.spec.architecture = runtime.GOARCH. However the runtime.GOARCH can vary depending where the virt-api pod is running.
1. In a multi-architecture cluster, assume that there are 10 worker nodes on which workloads can run, out of which 5 are arm64 and 5 are amd64. If the virt-api pod is running on arm64 first, the default will be arm64, for some reason it had to restart and it landed to amd64, then it will now be amd64? worse yet, if virt-api deployment creates each replica on a different architecture node, the default will depend on which replica serves the mutate request. If I am understanding correctly, this will be a very misbehaving default.
1. IIUC the above behavior is a side effect because of building a multi-arch image which is capable on running both amd64 and aarch64 runtimes.

Given this, I wonder if there is even a way to determine a default for the vmi.spec.architecture field in multi-architecture?

cc @rthallisey @rmohr for more conversation regarding correct defaults for this API.

Reply:
There are two ways out of this, 

first:
1. continue to have `vmi.spec.architecture` to be optional field. If not set, the value specified at `kubevirt.configuration.architectureConfiguration.defaultArchitecture` will be used for vmi.spec.architecture
1. make  `kubevirt.configuration.architectureConfiguration.defaultArchitecture` a  non-optional field
1. [optional] In order to not break upgrades, `kubevirt.configuration.architectureConfiguration.defaultArchitecture` can be backported to the current release,  and the field can be set as the `runtime.GOARCH` assuming that current release is run single architecture k8s deployments. Single architecture deployment assumption makes the default more sane. Users will then be expected to this upgrade kubevirt to the backport release (this will set the right defaults) before upgrading to release with multi-architecture support release. 

second:
1. make `vmi.spec.architecture` non-optional field. If not set, the value specified at `kubevirt.configuration.architectureConfiguration.defaultArchitecture` will be used for vmi.spec.architecture by the mutator. If vmi.spec.architecture is not specified by the user or is not mutated by the mutator, vmi is rejected.
1. make `kubevirt.configuration.architectureConfiguration.defaultArchitecture` optional field
1. [optional] In order to not break upgrades, `vmi.spec.architecture` can be backported to the current release,  and the field can be set as the `runtime.GOARCH` of mutator assuming that current release is run single architecture k8s deployments.

I personally like the first approach since I am assuming most users will be single architecture deployments and will not result in any breakage of current manifests upon upgrades.

Reply:
Hey @alaypatel07, I agree with you that option 1 more closely aligns with what already has been implemented and makes more sense. I will make the conversion of the field to be non-optional to avoid the scenario you described.

Reply:
Could you please elaborate a smidge further on your 3rd step in option 1? I'm not sure I understand entirely. Thanks.

Reply:
Hm, a third option would be to have the default archtecture mirrored to the KubeVirt CR status. We can then use runtime.GOARCH for existing installs to fill based on virt-operator the status once initially and take this value.|

My only request would be to not make a new field required, and  to not change spec fields automatically. I would consider both as an API breakage. The first because a new required API field is definitely an API breakage, and in the second case we would change the spec, potentially causing hotloops.

The fow would be like this:

 * If default architecture is unset in the spec, use runtime.GOARCH to determine the default (once) and set it in the status
 * If default architecture is set, still mirror it to the status
 * webhooks always falling back to the default architecture in the status if unset, no matter how the status got populated.

Would that work?

Reply:
I'm fine with the third option. @alaypatel07/ @rthallisey  any objections?

Reply:
Yes, it has now been implemented. Please review per our discussion